### PR TITLE
feat(type-tooltip): Add a tooltip with type information on mouse hover.

### PR DESCRIPTION
**Pull request by thso**
The tooltip appears when mousing over something in an editor and contains the same information as the title bar of `F1 dartdoc` information. It is currently hidden behind an experimental flag `hoverTooltip` set to `false` by default.

The tooltip in action:
tooltip_example

While roughly works, the functionality is here to be discussed and there is some work to be done on the content of the tooltip, the colors (hardcoded to blue and white right now), debouncing, and it currently shows even when the content is empty.

Linked to #7.

### CODE REVIEW

**Comment by devoncarew** [comment]
Thanks for the PR! I'll take a look at this a little later today.

File: styles/tooltip.less [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#hover-tooltip {
+  background-color: #226AEF;
```
Comment (devoncarew) [inline styles/tooltip.less]: We probably want to key off the current theme's style variables. There are a few examples in our less files.

Perhaps add a TODO: here, or track this in a bug to revise this in a follow-on PR.

File: lib/plugin.dart [hunk 2]
```diff
@@ -388,6 +390,14 @@ class AtomDartPackage extends AtomPackage {
         'type': 'boolean',
         'default': false,
         'order': 11
+      },
+
+      // experimental features
+      'hoverTooltip': {
```
Comment (devoncarew) [inline lib/plugin.dart]: We really only need to add config items here if we want them to be exposed to the user. We can still read (`atom.config.getValue()`) arbitrary keys w/o them being listed here.

Comment (devoncarew) [inline lib/plugin.dart]: We can keep this here though - just FYI.

**Comment by devoncarew** [comment]
This is great, thanks. We should track improvements in an issue. Before we remove the experimental flag, we'll want to:

- debounce, so that it shows when the mouse cursor is quiet for a short period
- borrow style (and content) from the current F1 dialog. It doesn't need to look the same; I think there's room for improvement in that dialog
- ensure that users are able to scroll within a tooltip w/ a lot of content w/o the tooltip disappearing when they move their mouse
- remove the F1 dialog
- remove the experimental flag :)

**Comment by devoncarew** [comment]
lgtm, landing. Can you open an issue w/ the above?

**Comment by thso** [comment]
Done, #825

**Pull request merged**